server/http/gateway/gateway.go

```diff
@@ -0,0 +1,41 @@
+package gateway
+
+import (
+	"net/http"
+
+	"github.com/gorilla/mux"
+
+	pluginapi "github.com/mattermost/mattermost-plugin-api"
+
+	"github.com/mattermost/mattermost-plugin-apps/apps"
+	"github.com/mattermost/mattermost-plugin-apps/server/appservices"
+	"github.com/mattermost/mattermost-plugin-apps/server/config"
+	"github.com/mattermost/mattermost-plugin-apps/server/proxy"
+	"github.com/mattermost/mattermost-plugin-apps/server/utils/httputils"
+)
+
+type gateway struct {
```

Comment:
J/C - why carving out the `gateway` package from the `restapi` one?

Reply:
This is a great question, I struggled with it myself.

On the one hand, I am very tempted to roll all the `http/*` packages into 1, esp. considering that "dialog" is a goner soon. On the other hand, there are 2 distinct buckets - proxying App interactions (gateway) and REST APIs, mostly for the Apps. There is some clarity in separation. IMO the `Call` API should become something like a POST to a `.../apps/AppID[/call]/PATH` and we may deprecate the `Call` REST API.

Long story short, 0/5 we either roll all http into 1 package (like we did proxy/admin), or we keep the 3 sub-packages for clarity?

Reply:
If we roll them, 1/5 a separate PR?